force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -126,16 +131,24 @@ public with sharing class ClassesObjectsMethods {
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
 
         // The variable to store the concatenated string
-
+        String finalNames = '';
+        
         // Loop through the list of scientists' names
+        for(String names : scientistNames){
+          // If the name doesn't contain the letter 'G', skip this iteration
 
-            // If the name doesn't contain the letter 'G', skip this iteration
-
-            // Add the name to the result string, followed by a comma
-
-        // Remove the last comma from the result string
-
-        return null; // Replace null with the variable you used to store the result
+        if(!names.toLowerCase().contains('g')){
+                continue;
+        }
+          // Add the name to the result string, followed by a comma
+            finalNames += names + ',';
+        }
+        
+          // Remove the last comma from the result string
+        if(!String.isBlank(finalNames)){
+        finalNames = finalNames.substring(0, finalNames.length() -1);
```

Comment:
Take a look at the String.join method. https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_string.htm#apex_System_String_join Could you use that here to make this easier?